### DISCUSSION TITLE
CLI Suggestions

### DISCUSSION THREAD

**Discussion by Leokuma** [Hexagon/pup#14] [Ideas]
I'm sure there was a reason for the current approach, but I just would like to suggest a few changes that I believe are common practice in terms of CLI usage.

Current | Proposed | Obs
------- | -------- | ---
`pup --help`  | `pup` | `--help` would still work as well.
`pup --start` | `pup start`
`pup`         | `pup start` | Currently it assumes `--start` (?). I think it would be more clear to be <br> explicit about it and require the command. Some people could type `pup` just <br> to see if it's installed or to see the commands available.
`pup --stop`  | `pup stop`
`pup --restart task-1`  | `pup restart task-1`

If those changes make sense, the same logic could be applied to other commands.

**Comment by Hexagon** [comment]
This makes perfect sense! 👍 

How about the configution helpers (init, append, remove), should these be individual actions, or grouped under (for example) "config"?

**Reply by Leokuma** [reply]
Hm good question. Both make sense and would work well for me. I don't have a strong opinion about it. Maybe you can decide this based on your view for the future of Pup.

**Reply by Hexagon** [reply]
I'll think about it 👍 

Grouping would make more sense than not if additional groups that require any of these keywords is expected. Hmm.

... maybe plugins could be added using the cli, with `pup plugin add`. Followin that pattern, `pup --append` would become `pup process add` rather than `pup config append`

**Comment by Hexagon** [comment]
Current plan (WIP):

```
# Pup without base argument
pup  

   No flags: Shows help and version, then exit

   --help
   --version
   --upgrade [<version>]

# Start pup

pup run

  (no arguments at all): pup will look for a configuration file and start, or exit if none were found

  --config|-c	Pup will use this configuration, or exit if not found

   OR ...	Use the process configuration flags to form a temporary configuration to run

  [--cmd|-C]
  [--autostart|-A]
  [--watch|-w]
  [--O|--cron]
  [-W|--cwd]

# Control running processes
pup

   start 	all|<proc-id>
   restart	all|<proc-id>
   block	all|<proc-id>
   unblock	all|<proc-id>
   stop		all|<proc-id>
   terminate	all|<proc-id>

  --config|-c	Use a configuration file other than default

# Get status of a running instance
pup 
  status
  [--config|-c]	Pup will use this configuration, or exit if not found

# Configuration file helpers
pup
   init|append|remove
```

**Reply by Leokuma** [reply]
LGTM 👍

**Comment by Hexagon** [comment]
Done in `1.0.0-beta.3` 🚀
